### PR TITLE
Feat: type safety in reducers/actions mixerconnections

### PR DESCRIPTION
**Pull request by olzzon**
Added type safety for:
* Fader reducers/actions
* Channel reducer/actions
* Settings reducer/actions
* Mixer connections
* Mixer connection types

It's over 1500 changes in the code, so a deep test is needed.

### CODE REVIEW

File: server/src/utils/mixerConnections/VMixMixerConnection.ts [hunk 1]
```diff
@@ -215,9 +234,11 @@ export class VMixMixerConnection {
                             outputLevel > 0 &&
                             Math.abs(outputLevel - input.volume) > 0.01
                         ) {
-                            dispatch(
-                                storeFaderLevel(assignedFaderIndex, input.volume)
-                            )
+                            this.dispatch({
```
Comment (mint-dewit) [inline server/src/utils/mixerConnections/VMixMixerConnection.ts:237]: This cannot be replaced outright, instead line 225 should be changed to be typed instead.

**Review by mint-dewit — CHANGES_REQUESTED** [review]
Great improvement! Would love to see the indexReducer typed as well i.e.

```
combineReducers<{
    faders: IFaders[],
    channels: IChannels[],
    settings: ISettings[]
}, FaderActions | ChannelActions | SettingsActions>
```

File: client/src/components/ChanStrip.tsx [hunk 2]
```diff
@@ -42,33 +40,54 @@ const DEL_VALUES = [10, 1, -1, -10]
 class ChanStrip extends React.PureComponent<
     IChanStripProps & IChanStripInjectProps & Store
 > {
+    dispatch: Dispatch<SettingsActions> = this.props.dispatch
```
[the remark this answers is not in the document]
Comment (olzzon) [reply client/src/components/ChanStrip.tsx:43]: Yes, I've just added the Types in the Store, by your suggest, and man i wish I'd done that before.
So I will clean up this

File: server/src/utils/mixerConnections/VMixMixerConnection.ts (hunk 1 above)
Comment (olzzon) [reply server/src/utils/mixerConnections/VMixMixerConnection.ts]: My fault, will add types, and rename dispatch to dispatchAndSetUpdateState. 
Didn't notice:
```
const dispatch = (update: any) => {
                    store.dispatch(update)
                    sendUpdate = true
                }

**Review by LindvedKrvang — APPROVED** [review]

**Review by RasmusAlbrektsen — APPROVED** [review]

**Pull request merged**